rusoto/core/src/region.rs

```diff
@@ -28,6 +28,7 @@ pub enum Region {
     UsWest1,
     UsWest2,
     CnNorth1,
+    Local(&'static str)
```

Comment:
I don't think we want to force people to use a static string. For example, if the endpoint was in a config file, this would not be possible to use (without some build script hackery). Consider using a general lifetime `&str` (this will be a pain because everything would have to be updated to use that lifetime) or just a `String`.